horace_core/sqw/@sqw/private/cut_accumulate_data_.m

```diff
@@ -46,71 +49,42 @@
     return
 end
 
-% Get the cumulative sum of pixel bin sizes and work out how many
-% iterations we're going to need
-cum_bin_sizes = cumsum(bin_ends - bin_starts);
 block_size = obj.data.pix.base_page_size;
-max_num_iters = ceil(cum_bin_sizes(end)/block_size);
+% Get indices in order to split the candidate bin ranges into pixel page sized
+% chunks
+[~, sub_bin_idxs] = split_vector(bin_ends - bin_starts, block_size);
+num_iters = size(sub_bin_idxs, 2);
```

Comment:
is this `num_chunks` or `num_iters`? The method is described as "paged sized chunks"